Adding support to skip tests based on provided xtf.<product ID>.subId

fabiobrz opened the pull request:
This PR is a proposal for adding support for skipping tests based on the `xtf.<productId>.subid` system property, as per the issue below, since WFLY 27 is not so far away...

Tests added as well, using https://mvnrepository.com/artifact/uk.org.webcompere/system-stubs-jupiter/2.0.1 which seems lightweight enough for use in test scope in order to set properties.

CC @mchoma

Fixes #470 

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

mnovak1 commented:
@fabiobrz I've just merged https://github.com/xtf-cz/xtf/pull/480 which affects the same code. Could you rebase it, please? I'm afraid there will be conflicts in `SkipFor` class which start to be more complex.

fabiobrz commented:
> @fabiobrz I've just merged #480 which affects the same code. Could you rebase it, please? I'm afraid there will be conflicts in `SkipFor` class which start to be more complex.

Hi @mnovak1 - thanks for the heads-up. Rebased and adjusted some tests.

simkam commented on junit5/src/main/java/cz/xtf/junit5/extensions/SkipForCondition.java:
```diff
@@ -51,11 +51,12 @@ public static ConditionEvaluationResult resolve(SkipFor skipFor) {
             DockerImageMetadata metadata = DockerImageMetadata.get(OpenShifts.master(), image);
             matcher = Pattern.compile(skipFor.imageMetadataLabelArchitecture()).matcher(metadata.labels().get("architecture"));
+        } else {
+            matcher = Pattern.compile(skipFor.subId()).matcher(XTFConfig.get("xtf.eap.subid"));
```
It needs to be somehow generic, not using `eap` id.

fabiobrz replied:
Thanks @simkam ! I must have lost it somewhere, fixing!

mnovak1 reviewed (approved):
LGTM.

mnovak1 commented:
Merging, thanks for the PR.

mchoma commented:
In my opinion this can replace `name`, `imageMetadataLabelName`, `imageMetadataLabelArchitecture` as all three options in the end serves to identify image in subId. Of course we need to handle backward compatibility.

So for example where `imageMetadataLabelArchitecture` was needed now `subId` can do work as well. in other words seems to me with this feature in place we do not need recent `imageMetadataLabelArchitecture` option.

dale-fu commented:
I do still have use for `imageMetadataLabelArchitecture` in other products as sometimes our team shares the same `subId` as x86.
